projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-plugins-modify-endpoint.php

```diff
@@ -346,6 +346,12 @@ protected function update() {
 				continue;
 			}
 
+			// Establish per plugin lock.
+			$plugin_slug = Jetpack_Autoupdate::get_plugin_slug( $plugin );
+			if ( ! WP_Upgrader::create_lock( 'jetpack_' . $plugin_slug ) ) {
+				continue;
```

Comment:
Should we track lock failures here? This is a request to update a plugin, yet we are not updating it because we cannot get the necessary lock. We cannot say that the update succeeded because we are not performing it.

Then again, we likely cannot get the lock because something else is trying the update.

Reply:
I don't think we have need to monitor failures to receive the lock as you mention not getting the lock means something else is updating the plugin so the desired action is still occurring.